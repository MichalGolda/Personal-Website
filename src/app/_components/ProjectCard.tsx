import Image from "next/image";

export type ProjectCardProps = {
  title: string;
  description: string;
  showUrl?: string;
  githubUrl?: string;
  coverImageUrl: string;
};

const ProjectCard: React.FC<ProjectCardProps> = ({
  title,
  description,
  showUrl,
  githubUrl,
  coverImageUrl,
}) => {
  return (
    <div className="flex flex-col gap-y-4">
      <h3 className="text-secondary font-bold text-3xl">{title}</h3>
      <div className="flex flex-row gap-x-4">
        {showUrl && (
          <a
            className="text-primary font-bold flex flex-row gap-x-4"
            href={showUrl}
          >
            Poka≈º
            <Image src="/arrow.svg" alt="arrow" width={32} height={0} />
          </a>
        )}
        {githubUrl && (
          <a
            className="text-primary font-bold flex flex-row gap-x-4"
            href={githubUrl}
          >
            Github
            <Image src="/arrow.svg" alt="arrow" width={32} height={0} />
          </a>
        )}
      </div>
      <p className="text-body">{description}</p>
      <Image className="!relative" src={coverImageUrl} alt={title} fill />
    </div>
  );
};

export default ProjectCard;
