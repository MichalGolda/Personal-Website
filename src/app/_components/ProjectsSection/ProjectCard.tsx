import Image from "next/image";
import { Project } from "@/app/_types/project";
import ProjectExternalLink from "./ProjectExternalLink";

export type ProjectCardProps = Project;

const ProjectCard: React.FC<ProjectCardProps> = ({
  name,
  description,
  previewUrl,
  githubUrl,
  coverImageSrc,
  ...props
}) => {
  return (
    <div className="flex flex-col gap-y-4" {...props}>
      <h3 className="text-secondary font-bold text-xl md:text-2xl lg:text-3xl">
        {name}
      </h3>
      <div className="flex flex-row gap-x-4">
        {previewUrl && <ProjectExternalLink name="PodglÄ…d" href={previewUrl} />}
        {githubUrl && <ProjectExternalLink name="Github" href={githubUrl} />}
      </div>
      <p className="text-body">{description}</p>
      <Image
        className="!relative object-contain !h-fit"
        src={coverImageSrc}
        alt={name}
        fill
      />
    </div>
  );
};

export default ProjectCard;
