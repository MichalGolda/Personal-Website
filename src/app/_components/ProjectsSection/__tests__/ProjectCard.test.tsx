import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import ProjectCard from "../ProjectCard";

describe("ProjectCard component", () => {
  it("renders without errors", () => {
    const { container } = render(
      <ProjectCard
        name="test name"
        description="test description"
        previewUrl="test preview url"
        githubUrl="test github url"
        coverImageSrc="/project.webp"
      />
    );

    const name = screen.getByText("test name");
    const links = screen.getAllByRole("link");
    const previewLink = links[0];
    const githubLink = links[1];
    const description = screen.getByText("test description");
    const coverImage = screen.getByAltText("test name");

    expect(name).toBeInTheDocument();
    expect(name.tagName).toBe("H3");

    expect(previewLink).toBeInTheDocument();
    expect(previewLink).toHaveAttribute("href", "test preview url");
    expect(previewLink.textContent).toBe("PodglÄ…d");

    expect(githubLink).toBeInTheDocument();
    expect(githubLink).toHaveAttribute("href", "test github url");
    expect(githubLink.textContent).toBe("Github");

    expect(description).toBeInTheDocument();
    expect(description.tagName).toBe("P");

    expect(coverImage).toBeInTheDocument();
    expect(coverImage).toHaveAttribute(
      "src",
      "/_next/image?url=%2Fproject.webp&w=3840&q=75"
    );
  });

  it("renders unchanged", () => {
    const { container } = render(
      <ProjectCard
        name="test name"
        description="test description"
        previewUrl="test preview url"
        githubUrl="test github url"
        coverImageSrc="/project.webp"
      />
    );

    expect(container).toMatchSnapshot();
  });
});
