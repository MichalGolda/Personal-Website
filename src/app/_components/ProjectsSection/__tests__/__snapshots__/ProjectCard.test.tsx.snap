// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectCard component renders unchanged 1`] = `
<div>
  <div
    class="flex flex-col gap-y-4"
  >
    <h3
      class="text-secondary font-bold text-xl md:text-2xl lg:text-3xl"
    >
      test name
    </h3>
    <div
      class="flex flex-row gap-x-4"
    >
      <a
        class="text-primary text-sm md:text-base font-bold flex flex-row gap-x-4"
        href="test preview url"
      >
        PodglÄ…d
        <img
          alt="arrow"
          data-nimg="1"
          decoding="async"
          height="0"
          loading="lazy"
          src="/arrow.svg"
          style="color: transparent;"
          width="32"
        />
      </a>
      <a
        class="text-primary text-sm md:text-base font-bold flex flex-row gap-x-4"
        href="test github url"
      >
        Github
        <img
          alt="arrow"
          data-nimg="1"
          decoding="async"
          height="0"
          loading="lazy"
          src="/arrow.svg"
          style="color: transparent;"
          width="32"
        />
      </a>
    </div>
    <p
      class="text-body"
    >
      test description
    </p>
    <img
      alt="test name"
      class="!relative object-contain !h-fit"
      data-nimg="fill"
      decoding="async"
      loading="lazy"
      sizes="100vw"
      src="/_next/image?url=%2Fproject.webp&w=3840&q=75"
      srcset="/_next/image?url=%2Fproject.webp&w=640&q=75 640w, /_next/image?url=%2Fproject.webp&w=750&q=75 750w, /_next/image?url=%2Fproject.webp&w=828&q=75 828w, /_next/image?url=%2Fproject.webp&w=1080&q=75 1080w, /_next/image?url=%2Fproject.webp&w=1200&q=75 1200w, /_next/image?url=%2Fproject.webp&w=1920&q=75 1920w, /_next/image?url=%2Fproject.webp&w=2048&q=75 2048w, /_next/image?url=%2Fproject.webp&w=3840&q=75 3840w"
      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
    />
  </div>
</div>
`;
